databricks/resource_databricks_permissions_test.go

```diff
@@ -0,0 +1,142 @@
+package databricks
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/databrickslabs/databricks-terraform/client/model"
+	"github.com/databrickslabs/databricks-terraform/client/service"
+)
+
+func TestAccDatabricksPermissionsResourceFullLifecycle(t *testing.T) {
+	var permissions model.ObjectACL
+	randomName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// create a resource
+				Config: testClusterPolicyPermissions(randomName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("databricks_permissions.dummy_can_use",
+						"object_type", "cluster-policy"),
+					testAccIDCallback(t, "databricks_permissions.dummy_can_use",
+						func(client *service.DBApiClient, id string) error {
+							resp, err := client.Permissions().Read(id)
+							if err != nil {
+								return err
+							}
+							permissions = *resp
+							assert.Len(t, permissions.AccessControlList, 3)
+							return nil
+						}),
+				),
+			},
+			{
+				Config: testClusterPolicyPermissionsSecondGroupAdded(randomName),
+				Check: testAccIDCallback(t, "databricks_permissions.dummy_can_use",
+					func(client *service.DBApiClient, id string) error {
+						resp, err := client.Permissions().Read(id)
+						if err != nil {
+							return err
+						}
+						permissions = *resp
+						assert.Len(t, permissions.AccessControlList, 3)
+						return nil
+					}),
+			},
```

Comment:
We should have a test of the permissions being deleted manually, or the parent resource deleted manually and the plan should not be empty and not throw an error.

Reply:
you mean the isMissing in read resource?